### PR TITLE
Move generation of cha_commitid.c to Makefile

### LINKED ISSUE #62: Make doesn't update commit ID after a git pull
With the new build system, the cha_commitid.c file with the git version is created at configure time.  The Makefile doesn't test for a change in git versions, so a sequence like:
git pull; make clean; make
will result in an executable that reports an incorrect commit ID.

### PR DESCRIPTION
**Pull request by hainest**
Fixes #62.

### CODE REVIEW

**Comment by hainest** [comment]
Passes basic build test on Comet.

File: Makefile.in [hunk 1]
```diff
@@ -267,6 +267,11 @@ $(build_dir)/libmodule%.a: $(build_dir)/%.o
 	@ echo Creating library $@...
 	$(quiet) $(AR) $@ $^
 
+.PHONY: git_commit
+git_commit:
+	$(quiet) VER=$(cd $srcdir && git describe --long --always)
```
Comment (trquinn) [inline Makefile.in:272]: I don't think $srcdir is the right variable here.  $source_dir maybe?

File: Makefile.in [hunk 2]
```diff
@@ -267,6 +267,11 @@ $(build_dir)/libmodule%.a: $(build_dir)/%.o
...
+	$(quiet) echo "const char * const Cha_CommitID = \"$VER\";" > cha_commitid.c
```
Comment (trquinn) [inline Makefile.in:273]: This produces a file with:
const char * const Cha_CommitID = "ER";

Comment (hainest) [reply Makefile.in]: Well, that's neat. My local version contained the required changes to make this work, but I evidently never added them to the cache to commit them. Thanks for double-checking this!

**Comment by hainest** [comment]
cha_commitid.c isn't being generated in time to get picked up for the source list. It's a quick fix. I'll run it through the build tests again, as well.

**Pull request merged**